packages/matchbox/src/components/Popover/Popover.js

```diff
@@ -1,159 +1,153 @@
-    bottom: PropTypes.bool,
-    /**
-      * Callback function that is called when clicking outside the popover, or hitting escape.
-      */
-    onClose: PropTypes.func,
-    /**
-      * Popover Content
-      */
-    children: PropTypes.node,
-    /**
-     * Element ID for the portal that will house Popover. Appends to body if not provided.
-     */
-    portalId: PropTypes.string
-  };
-
-  static defaultProps = {
-    right: true,
-    bottom: true
-  }
+function Popover(props) {
+  const { as, id, open: controlledOpen, onClose, children, trigger, wrapper, ...rest } = props;
+  const [open, setOpen] = React.useState(null);
+  const popoverRef = React.useRef();
+  const activatorRef = React.useRef();
 
-  state = {
-    open: null
-  }
+  const shouldBeOpen = controlledOpen || open;
+  const Wrapper = as || wrapper || 'span';
 
-  componentDidMount() {
-    const { open: controlledOpen } = this.props;
+  useWindowEvent('click', handleOutsideClick);
+  useWindowEvent('keydown', handleEsc);
 
+  // Mount hook – sets initial close state if uncontrolled
+  React.useEffect(() => {
     // This component becomes "controlled" if the prop 'open' is given a boolean value
     if (controlledOpen === undefined) {
-      this.setState({ open: false });
+      setOpen(false);
     }
-  }
+  }, []);
 
-  uncontrolledToggle = () => {
-    this.setState({ open: !this.state.open });
+  // Toggles uncontrolled open state
+  function handleUncontrolledToggle() {
+    setOpen(!open);
   }
 
-  handleOutsideClick = (e) => {
-    const { open: controlledOpen, onClose } = this.props;
-    const isOutside = this.popover && !this.popover.contains(e.target) && this.activator && !this.activator.contains(e.target);
+  // Toggles uncontrolled popovers, and calls `onClose` for controlled popovers
+  function handleOutsideClick(e) {
+    const isOutside =
+      !popoverRef.current.contains(e.target) && !activatorRef.current.contains(e.target);
 
-    if (isOutside && (controlledOpen || this.state.open)) {
+    if (isOutside && shouldBeOpen) {
       if (onClose) {
         onClose(e);
       }
 
-      if (this.state.open) {
-        this.uncontrolledToggle();
+      if (open) {
+        handleUncontrolledToggle();
       }
     }
   }
 
-  handleEsc = (e) => {
-    const { open: controlledOpen, onClose } = this.props;
-
-    if (onClose && (controlledOpen || this.state.open)) {
+  // Toggles uncontrolled popovers, and calls `onClose` for controlled popovers
+  function handleEsc(e) {
+    if (onClose && shouldBeOpen) {
       onKey('escape', onClose)(e);
     }
 
-    if (this.state.open) {
-      onKey('escape', this.uncontrolledToggle)(e);
+    if (open) {
+      onKey('escape', handleUncontrolledToggle)(e);
     }
   }
 
-  handleTrigger = () => {
-    if (this.state.open === false) {
-      this.uncontrolledToggle();
+  // Toggles uncontrolled popovers
+  function handleTrigger() {
+    if (open !== null) {
+      handleUncontrolledToggle();
     }
```

Comment:
A few things about the trigger:
1. It looks like `aria-expanded` is missing from the example on the button element...I know this won't be supplied by the component (though I think ideally it would - maybe when we update component APIs)
1. Clicking on the button repeatedly appears to cause a bug? It sort of opens and closes but appears to get stuck in a certain state, especially if you click rapidly.

Reply:
1. This was addressed.
2. I saw the same thing.  Jon thinks it is due to Storybook.  Punting for now.